src/Pos/Block/Network/Listeners.hs

```diff
@@ -94,3 +101,22 @@ handleBlockHeaders = listenerConv @MsgGetHeaders $ \__ourVerInfo nodeId conv ->
     mHeaders <- recvLimited conv
     whenJust mHeaders $ \(MsgHeaders headers) ->
         handleUnsolicitedHeaders (getNewestFirst headers) nodeId
+
+----------------------------------------------------------------------------
+-- Subscription
+----------------------------------------------------------------------------
+
+handleSubscription
+    :: forall ssc ctx m.
+       (WorkMode ssc ctx m)
+    => (ListenerSpec m, OutSpecs)
+handleSubscription = listenerConv @Void $ \__ourVerInfo nodeId conv -> do
+    mbMsg <- recvLimited conv
+    whenJust mbMsg $ \MsgSubscribe -> do
+      -- TODO: Should we change this so that MsgSubscribe carries the node type?
+      -- In particular, for the "transitional" mode where Kademia nodes
+      -- use this subscription mechanism too?
```

Comment:
No :-), but we should have this depend on configuration on the listener side, ie when in transitional mode this listener should consider all subscribing peers to be core.